test/assert_relationship_test.exs

```diff
@@ -99,4 +99,9 @@ defmodule AssertRelationshipTest do
         assert "record with `id` 1 and `type` \"post\" was found but had mis-matching attributes" == error.message
     end
   end
+
+  test "will return the original payload" do
+    payload = assert_relationship(data(:payload), data(:author), as: "author", for: data(:post))
```

Comment:
We should consider an assert api similar to phoenix test helpers where we use the `assert` macro so we can accomplish two checks in one pass. So my proposal would be to support this:

    assert relationship(data(:payload), data(:author), as: "author", for: data(:post)) == data(:payload)

Thoughts?